src/subword.rs

```diff
@@ -157,13 +157,85 @@ impl SubwordIndices for str {
     }
 }
 
+/// Create an Iterator going through a sentence
+///
+/// NGramsIndicesIter provides an iterator produces n-grams in a sentence between a
+/// minimum and maximum length and the indices of the ngrams in form of (ngram, index).
+/// "ngram" if of type &str, and index is of type u64.
+///
+/// **Warning:** no guarantee is provided with regard to the iteration
+/// order. The iterator only guarantees that all n-grams and their indices are produced.
+struct NGramsIndicesIter<'a> {
+    ngrams: NGrams<'a>,
+    mask: u64,
+}
+
+impl<'a> NGramsIndicesIter<'a> {
+    pub fn new(string: &'a str, min_n: usize, max_n: usize, buckets_exp: usize) -> Self {
+        assert!(
+            buckets_exp <= 64,
+            "The largest possible buckets exponent is 64."
+        );
+
+        let mask = if buckets_exp == 64 {
+            !0
+        } else {
+            (1 << buckets_exp) - 1
+        };
+
+        let ngrams = NGrams::new(string, min_n, max_n);
+        NGramsIndicesIter { ngrams, mask }
+    }
+}
+impl<'a> Iterator for NGramsIndicesIter<'a> {
+    type Item = (&'a str, u64);
+
+    #[inline]
+    fn next(&mut self) -> Option<Self::Item> {
+        let next_ngram = self.ngrams.next();
+        match next_ngram {
+            Some(ngram) => {
+                let mut hasher = FnvHasher::default();
+                ngram.hash(&mut hasher);
+                Some((ngram.inner, hasher.finish() & self.mask))
+            }
+            None => return None,
```

Comment:
Superfluous return, since this is the last expression of `next`, this can be

```
None => None,
```

However, since you only have to do something in the `Some` case, the function body can be simplified to:

```rust
self.ngrams.next().map(|ngram| {
  let mut hasher = FnvHasher::default();
  ngram.hash(&mut hasher);
  (ngram.inner, hasher.finish() & self.mask)
}
```